### PR TITLE
[Recreated PR] Threads should not refresh from server when navigating to from discussions page

### PR DESCRIPTION
**Pull request by mzparacha**
Recreation of https://github.com/hicommonwealth/commonwealth/pull/3632

### CODE REVIEW

File: packages/commonwealth/server/routes/getThreads.ts
Comment (NakulManchanda) [inline packages/commonwealth/server/routes/getThreads.ts:42]: this created OOM errors.
https://github.com/hicommonwealth/commonwealth/issues/3950

As you mentioned on slack, you are just looking for comment count, we have util function already implemented here:
https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/server/util/getThreadCommentsCount.ts

Comment (mzparacha) [reply packages/commonwealth/server/routes/getThreads.ts:42]: Hi, Thanks for the heads up @NakulManchanda. Will definitely give it a go today.

Comment (NakulManchanda) [reply packages/commonwealth/server/routes/getThreads.ts]: update looks good to me, thanks!

**Comment by NakulManchanda** [comment]
@alexyoung23j can you add example for https://github.com/hicommonwealth/commonwealth/pull/3587 here?
We bought the commented code back for now, would think of different solution if really needed!
I would like to try with redirected request.

**Comment by NakulManchanda** [comment]
We need to satisfy this too @mzparacha 
https://commonwealth.im/discussion/11224 
https://github.com/hicommonwealth/commonwealth/issues/3584
Can you test it once?

File: packages/commonwealth/server/routes/getThreads.ts
Comment (NakulManchanda) [inline packages/commonwealth/server/routes/getThreads.ts:48]: change to 
`chainId: chain?.id || threads[0].chain,` to satisfy this https://github.com/hicommonwealth/commonwealth/pull/3959#issuecomment-1562242248

File: packages/commonwealth/server/routing/router.ts
Comment (NakulManchanda) [inline packages/commonwealth/server/routing/router.ts:484]: comment this back to preserve previous fix - verify with @jnaviask

**Review by NakulManchanda — CHANGES_REQUESTED** [review]

**Review by jnaviask — CHANGES_REQUESTED** [review]
The overall issue here, of adding sequelize associations without proper database support, needs addressing before we can merge this PR.

**Comment by NakulManchanda** [comment]
@jnaviask 
> adding sequelize associations without proper database support, needs addressing before we can merge this PR.
is there any pending issue?
Can you highlight one in particular for this PR?

@mzparacha let me know if you think someone from backend should help here. Thanks!

File: packages/commonwealth/server/routes/getThreads.ts
Comment (mzparacha) [reply packages/commonwealth/server/routes/getThreads.ts:48]: no sure if it will work when thread ids from different communities are passed in query params - i think it won't ideally happen tho.

Comment (NakulManchanda) [reply packages/commonwealth/server/routes/getThreads.ts:48]: i don't thin we allow it intentionally though
- to be safe you can add for-loop - for all different chain_ids!
- but would be a overkill i believe!

**Comment by mzparacha** [comment]
Thanks for looking into this @NakulManchanda.

How about we pass the chainId conditionally for `getThreadsWithCommentCount`, so something like changing this line
https://github.com/hicommonwealth/commonwealth/blob/bc6e7bcc92217b53e09c83d8f3c73732bc1617d5/packages/commonwealth/server/util/getThreadCommentsCount.ts#L21

to 

```jsx
...(chainId && { chain: chainId }),
```

and implement [this](https://github.com/hicommonwealth/commonwealth/pull/3959#discussion_r1204984967) change as you suggested.

@jnaviask, just to confirm if you agree with these changes?

**Comment by NakulManchanda** [comment]
yeah, thats a very good idea much better @mzparacha 
WE don't need to uncomment validate chain as well

we pass the chainId conditionally for getThreadsWithCommentCount, so something like changing this line
- just construct where clause in two steps:

```
where_clause = {
thread_id: { [Op.in]: rootIds },
      deleted_at: null,
}

if(chainId) {
where_clause[chain] = chainId
}

```

**Comment by jnaviask** [comment]
I can't follow the exact proposal for changes. Can someone write up the proposed changes without referring to outside context (other comments, other PRs, lines of code, etc)? I just want to see the entire plan in one place before approving anything.

**Comment by mzparacha** [comment]
@jnaviask, i think me and @NakulManchanda landed on these changes (@NakulManchanda plz correct me if wrong)

For some background (i know everyone already knows this just restating), the pr i created originally returned all comments for a thread from the /getThreads route, but in the frontend we only needed the count of comments. @NakulManchanda suggested to use the [`getThreadsWithCommentCount()`](https://github.com/hicommonwealth/commonwealth/blob/bc6e7bcc92217b53e09c83d8f3c73732bc1617d5/packages/commonwealth/server/util/getThreadCommentsCount.ts) utility to only return the comments count which we already did but this util requires the chain id.


ATM the suggested changes are to 
- use the `getThreadsWithCommentCount` in `getThreads` endpoint to return comments count (already added in a recent commit)
- **[ACTION ITEM]** remove validate chain from the getThreads api - basically undo [this](https://github.com/hicommonwealth/commonwealth/pull/3959/files/60dcccbe3eeed87e48b1715d8b64c350e34053f0#diff-8d32fb76a63501c8bc5a195c11c61461f77df9e681aa981c010aa828b8f47137R484) 1 line change by commenting it
- **[ACTION ITEM]** make the chainId param for `getThreadsWithCommentCount` optional (it works perfectly without that too) - this is the final thing we have to do

hope i did not make it too wordy, just wanted to sum up the discussion till here!

**Comment by jnaviask** [comment]
@mzparacha Not at all too wordy, and super helpful, thanks!

What's the reason why we need to remove validate chain from getThreads? If it's because getThreads no longer takes a chain, what's the reason for removing the chain field?

**Comment by mzparacha** [comment]
@jnaviask validate chain was already removed from getThreads. 

image

I added it back to make the `getThreadsWithCommentCount ` work, but we figured out that the util can work with validate chain so we can undo my changes (remove it again)

**Comment by jnaviask** [comment]
Great, sounds good. Thanks for the clarification!

**Comment by mzparacha** [comment]
@jnaviask so do i have the green light to implement the change?

**Comment by NakulManchanda** [comment]
Tested on frick:
Url which broke the prod is fine now
https://commonwealth.im/stargatetoken/discussion/4353-proposal-guidelines

image

**Review by NakulManchanda — APPROVED** [review]
New changes don't produce OOM error on server, tested on Frick

**Comment by mzparacha** [comment]
thanks for the review @NakulManchanda!

File: packages/commonwealth/server/models/thread.ts
Comment (jnaviask) [inline packages/commonwealth/server/models/thread.ts:171]: Do we still need to add this?

Comment (mzparacha) [reply packages/commonwealth/server/models/thread.ts:171]: Not necessary, we can remove. This however won't do anything and there is still a relation b/w comments and threads from the db POV. Lmk if i should remove!

Comment (jnaviask) [reply packages/commonwealth/server/models/thread.ts:171]: Please remove them until we implement a formal association (right now it's only an implicit contract rather than a proper relation at the schema level).

Comment (NakulManchanda) [reply packages/commonwealth/server/models/thread.ts:171]: @jnaviask was just checking on schema for. different clean up ticker

We do have:
```
    "Comments_thread_id_fkey" FOREIGN KEY (thread_id) REFERENCES "Threads"(id)
```

image

Comment (jnaviask) [reply packages/commonwealth/server/models/thread.ts:171]: Came from #3996.

**Review by CowMuon — APPROVED** [review]
LGTM

**Review by jnaviask — APPROVED** [review]

**Pull request merged**
